Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -656,8 +674,19 @@ public final class SwiftTargetBuildDescription {
         result += self.buildParameters.sanitizers.compileSwiftFlags()
         result += ["-parseable-output"]
         result += try self.buildSettingsFlags()
+
         result += self.buildParameters.toolchain.extraFlags.swiftCompilerFlags
-        result += self.buildParameters.swiftCompilerFlags
+        // User arguments (from -Xswiftc) should follow generated arguments to allow user overrides
+        result += self.buildParameters.flags.swiftCompilerFlags
+
+        result += self.buildParameters.toolchain.extraFlags.cCompilerFlags.asSwiftcCCompilerFlags()
+        // User arguments (from -Xcc) should follow generated arguments to allow user overrides
+        result += self.buildParameters.flags.cCompilerFlags.asSwiftcCCompilerFlags()
+
+        result += self.buildParameters.toolchain.extraFlags.cxxCompilerFlags.asSwiftcCXXCompilerFlags()
```

Comment:
@compnerd @MaxDesiatov I'd really like to get some feedback here. I'm not sure if passing `-Xcxx flags` to swiftc is ok for all swift targets, but it seems maybe reasonable to do because the previous implementation passed the `-Xcc flags` (via buildParameters.swiftCompilerFlags).

AFAICT by default the only change is a new trailing arg to swiftc: `swiftc <old args> -Xcc -lc++` when compiling a an object this shouldn't matter and when linking I _assume_ the linker doesn't eager link libraries if no symbols from it are used.

Reply:
After reading #4994 I _think_ the cxxCompiler flags should go to swiftc prefixed by `-Xcxx` however this option does not exist in driver yet.

Reply:
I think in absence of `-Xcxx` we probably shouldn't forward the flags here, but if needed we could add *another* set of flags which are the CXX flags that are "safe" to pass to SwiftPM.

Reply:
I talked to @hyp this morning and it seems like we will be adding an -Xcxx to swift-driver some time in the future. When that happens we can start propagating spm's cxx flags to swift-driver.